hermes/src/wormhole.rs

```diff
@@ -141,15 +161,20 @@ pub async fn verify_vaa<'a>(
         (guardian_set.keys.len() * 2) / 3 + 1
     };
 
-    if num_correct_signers < quorum {
-        return Err(anyhow!(
+    // Check if we have enough correct signatures and update the header to
+    // contain only the minimal required signatures (quorum) to save gas
+    // upon VAA verification.
+    match valid_signatures.get(0..quorum) {
+        Some(valid_signatures) => {
+            header.signatures = valid_signatures.to_vec();
+            Ok((header, body).into())
+        }
+        None => Err(anyhow!(
```

Comment:
There's a few changes you can make to this function to make this a little clearer, remove the mutability, and flatten the flow a little bit to make it easier to follow (always nice to help the compiler with immutability optimizations), feel free to ignore as It's no big deal but up to you if you [like anything about this flow](https://pape.rs/2uMGXrdd) or to keep it the way it currently is, either is fine.